master/usage/ipv6.md

```diff
@@ -1,20 +1,155 @@
+	the host.
+  - The `--advertise-address` flag should be set to the IPv6 address that
+    nodes should use to access the kube-apiserver.
+  - The `--service-cluster-ip-range` flag should be set to an IPv6 CIDR
+    that will be used for the Service IPs, the DNS service address must
+	be in this range.
+- For kube-controller-manager:
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - If the `--allocate-node-cidrs` flag is set then it is necessary to set
+    `--node-cidr-mask-size`. These are necessary when using host-local IPAM,
+    but if using calico-ipam is is easier to remove both flags as they are
+    not needed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+- For kube-scheduler:
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+- For kubelet:
+  - The `--address` flag should be set to the appropriate IPv6 address
+    or `::` for all IPv6 addresses.
+  - The `--cluster-dns` flag should be set to the IPv6 address that will
+    be used for the service DNS, this must be in the range used for
+	`--service-cluster-ip-range`.
+  - The `--node-ip` flag should be set to the IPv6 address of the node.
+- For kube-proxy:
+  - The `--bind-address` flag should be set to the appropriate IPv6
+    address or `::` for all IPv6 addresses on the host.
+  - The `--master` flag should be set with the IPv6 address where the
+    kube-apiserver can be accessed.
+  - The `--cluster-cidr` flag must be set to match the {{site.prodname}}
+    IPv6 IPPool.
+
+> **Note:** These flags are not the only flags that will need to be set on the
+> kubernetes components but the ones that were found necessary to ensure IPv6
+> operation.
+{: .alert .alert-info}
+
+### Calico Manifest changes
+
+{{site.prodname}} manifests can be modified to work in an IPv6
+deployment by making the following changes.
+
+- If the ipam section of the `cni_network_config` has
+  `"type": "calico-ipam"` then it should be modified to have
+  [IPv4 assigning disabled and IPv6 assigning
+  enabled](/{{page.version}}/reference/cni-plugin/configuration#ipam).
+- Add the following environment variables to the calico-node Daemonset
+
+  ```
+  - name: CALICO_IPV6POOL_CIDR
+    value: "fd20::0/112"
+  - name: IP6
+    value: "autodetect"
+  ```
+
+  - Set the `CALICO_IPV6POOL_CIDR` to the desired pool, it should match
+    the `--cluster-cidr` passed to the kube-controller-manager and to
+	kube-proxy.
+- Ensure the environment variable `FELIX_IPV6SUPPORT` is set `true`
+  on the calico-node Daemonset.
+
+### DNS changes
+
+It will probably be necessary to modify your DNS pod for IPv6. If you are using
+[KubeDNS](/{{page.version}}/getting-started/kubernetes/installation/manifests/kubedns.yaml), 
+then the following changes will ensure IPv6 operation.
+
+- Update the image versions to at least `1.14.8`.
```

Comment:
It would be great if there was upstream Kubernetes documentation we could point to for all of this.... I'm guessing nothing exists for now but when it does we can clean this up a lot.